### PR TITLE
Encrypted samples component test

### PR DESCRIPTION
**Pull request by skywilliamsonl**
Summary: Encrypted audio video playback component test.
Type: Feature
Test Plan: Unittests, Component Tests.
Jira: RIALTO-436

### CODE REVIEW

File: tests/componenttests/client/tests/base/MediaPipelineTestMethods.cpp [hunk 1]
```diff
@@ -409,6 +415,90 @@ void MediaPipelineTestMethods::checkMseVideoSegmentWrittenSecondary(int32_t segm
...
+void MediaPipelineTestMethods::checkEncryptedVideoSegmentWritten(int32_t segmentId, uint32_t keyIndex)
+{
+    auto it = writtenVideoSegments.find(segmentId);
+    ASSERT_NE(it, writtenVideoSegments.end());
+
+    uint8_t *dataPosition{reinterpret_cast<uint8_t *>(getShmAddress()) + it->second.mediaDataOffset};
+    std::uint32_t *metadataSize{reinterpret_cast<uint32_t *>(dataPosition)};
+    dataPosition += sizeof(uint32_t);
+    MediaSegmentMetadata metadata;
+    ASSERT_TRUE(metadata.ParseFromArray(dataPosition, *metadataSize));
```
Comment (skyproudmanp) [inline tests/componenttests/client/tests/base/MediaPipelineTestMethods.cpp:491]: Should lines 445-449 and 487-491 (which are the same) become a function?

**Review by skyproudmanp — CHANGES_REQUESTED** [review]

File: tests/common/protoUtils/MetadataProtoUtils.h [hunk 2]
```diff
@@ -44,4 +44,31 @@ convertSegmentAlignment(const firebolt::rialto::MediaSegmentMetadata_SegmentAlig
...
+    case firebolt::rialto::CipherMode::CBCS:
+    {
+        return firebolt::rialto::MediaSegmentMetadata_CipherMode_CBCS;
+    }
+    case firebolt::rialto::CipherMode::UNKNOWN:
+    {
+        return firebolt::rialto::MediaSegmentMetadata_CipherMode_UNKNOWN;
+    }
+    }
+    return firebolt::rialto::MediaSegmentMetadata_CipherMode_UNKNOWN;
```
[the remark this answers is not in the document]
Comment (skyproudmanp) [reply tests/common/protoUtils/MetadataProtoUtils.h:72]: (The final "return" statement is also the default behaviour therefore you shouldn't feel obliged)

**Review by aczs — APPROVED** [review]

File: tests/componenttests/client/tests/base/MediaPipelineTestMethods.cpp (hunk 1 above)
Comment (skywilliamsonl) [reply tests/componenttests/client/tests/base/MediaPipelineTestMethods.cpp]: done

File: tests/common/protoUtils/MetadataProtoUtils.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (skywilliamsonl) [reply tests/common/protoUtils/MetadataProtoUtils.h:72]: done

**Review by aczs — APPROVED** [review]

**Review by skyproudmanp — APPROVED** [review]

**Review by aczs — APPROVED** [review]

**Pull request merged**